### ISSUE TITLE
Colab notebook not using GPU despite having selected GPU in runtime panel

### ISSUE THREAD

**Issue by frigeriomtt** [googlecolab/colabtools#2979]
I already exposed the problem in [this issue](https://github.com/googlecolab/colabtools/issues/2964). Despite enclosing my tf code with 

```python 
with tf.device('/device:GPU:0'):
```

like showed in [this colab tutorial](https://colab.research.google.com/notebooks/gpu.ipynb#scrollTo=Y04m-jvKRDsJ)  (which was proposed in the [answer link ](https://research.google.com/colaboratory/faq.html#gpu-utilization) of the previous issue), the resources tab still shows 0 usage of gpu resources (GPU is selected as runtime, of course).

**Comment by cperry-goog** [comment]
Can you share a minimal repro notebook?

**Comment by frigeriomtt** [comment]
@cperry-goog  I'm sharing with you a [directory](https://drive.google.com/drive/folders/1D_BOo6ozkoBwPNXDRXTfQpQTOZt0T59g?usp=sharing) with both the reproducible code and the dataset (eventually ask for access rights); you'll have to adjust the path strings in accord with the  structure of your gdrive (I suggest you to add a shortcut of the 'shared with keras' folder from the 'shared with me' directory to the MyDrive one, doing so the script should work as it is, minus the described problem); the notebook with the gpu usage theoretically handled is the one with gpu in the file name

**Comment by frigeriomtt** [comment]
> Can you share a minimal repro notebook?

@cperry-goog  you have been provided access.

**Comment by cperry-goog** [comment]
Sorry it took me a bit to get to this. Thanks for giving me good steps here!

I saw what you saw: 0 GPU usage, but then I got a Drive I/O error on your _180 folder; looks like you've got a ton of images in that folder and I'm betting you're running up against https://research.google.com/colaboratory/faq.html#drive-timeout

Did you get through without a Drive error? I'm guessing this is a Tensorflow bug if so, and we'd be unable to diagnose without involving them.
